Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -137,6 +161,13 @@ private PropertyInfo FindProperty(string readName)
             var name = options.NamingConvention.ToProperty(readName);
             return targetType.GetProperty(name);
         }
+
+        private bool IsReadOnlyProperty(PropertyInfo prop)
+        {
+            return !prop.CanWrite || 
```

Comment:
I think this is still incorrect. I think you should just remove the `!prop.CanWrite` part.

Reply:
After removing `!prop.CanWrite`, a property with a `{ get; }` accessor does not pass `IsReadOnlyProperty()` because it fails the `ReadOnlyAttribute` check.

So now I've redefined this to `property.SetMethod == null`.